### ISSUE TITLE
Bug: Database Error when entering observation

### ISSUE THREAD

**Issue by itravers** [MarineAppliedResearch/MARP_API#28]
Monday Tracking: https://mare658599.monday.com/boards/5542763614/pulses/7659048784

**Comment by itravers** [comment]
Hi Isaac,Thanks for the updated GUI. It is working a little better. However, I have had it not put in data after I hit a button. It seems like the same problem just now it never puts in a blank line. It does skip the Obs ID. For example, on dive 19 line 30 I hit the UI Henricia button and it did not show up. It would have been Obs ID 183. I tried it again and it worked the second time but went in as Obs ID 184. Attached is the log.
Also, the GUI is still running slow. We are trying to go at its slower speed so as to not make any errors related to that. It seems to run faster when only one of us is using it at a time.

Additionally, when Sam was cleaning the fish data he found some errors I made. I think that I may have edited them when processing but the edits didn't save into the final database. I put in the wrong fish size, then changed the size. The size looked like it had been changed correctly but when we went back into the database later it had not updated to the new size. For now we are trying not to edit anything in the GUI. Insted we delete the entry and put in a new correct one.
Thanks for all your help with it,
Johnathan Centoni |Biologist-Video Analyst


Replied:


Thank you for the detailed report! I have a good idea of what might be going wrong, but I need to see it happen live to fully understand and fix the issue.

From what I’ve observed, when only one person is using the system, it seems to work fine. But when multiple people use it at the same time, it starts to slow down. I traced the slowdown to how the software communicates with the backend (the server or database). Occasionally, when the software asks the backend for information, the backend doesn’t return any data.

The issue with the blank line happened because the backend sent back an empty response, but the software was expecting actual data. As a result, it created a new row but couldn’t fill it with information because there was none, so the row remained blank.

The next step is to figure out why the backend is sending empty data. To do that, I’ll need to be logged in and watching the system when the issue happens, which might be tricky to catch. My guess is that one of the database queries I wrote takes longer to run now that there’s more data in the system, which is why this problem wasn’t noticeable before but has gotten worse over time. Once I can identify the slow query, I should be able to optimize it and fix the problem.

Thanks again for your patience, and I’ll keep working on it!

Best,
Isaac
